### DISCUSSION TITLE
make error/warning more verbose

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [elchininet/custom-sidebar#242] [Ideas]
<img width="1279" alt="Scherm­afbeelding 2024-11-29 om 12 58 40" src="https://github.com/user-attachments/assets/75adca58-fa61-4598-9754-ee09433e9a05">

would help if the error could point to the actual config it references

I managed to find it though:
```
subtitle: >
  [[[ @partial subtitle
      return marquees != 0 ? phrase + marquees : none; ]]]
```

isn't excepted, and it requires to set an empty string apparently:
```
subtitle: >
  [[[ @partial subtitle
      return marquees != 0 ? phrase + marquees : ''; ]]]
```
or made I should not use the ternary template here

But besides that, it would be nice to have a better pointer to the actual perpetrator in the config. (maybe a debug setting?)

**Comment by elchininet** [comment]
Hi @Mariusthvdb,
The code that you write in the templates is not a real code compiled by a tool that can tell you from which line/file it is coming. It is an error coming from a [Function constructor instance](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/Function) in memory in which the code is evaluated dynamically, so it doesn't know from which line/file it comes because in reality it is not coming from there.

Remember that this is not an integration that runs in back-end, this is just a `JavaScript` plugin, so all the code runs in memory.

Anyway the error message will be there, if you find an error that says that there is no a variable called `none`, first thing to debug is to search for a `none` variable in your code. (In `JavaScript` if you write `something` without enclosing it in quotes, it is interpreted as a variable that you defined previously, and if it is not defined it will throw).
